src/jaxsim/parsers/descriptions/link.py

```diff
@@ -40,13 +40,15 @@ class LinkDescription(JaxsimDataclass):
 
     def __hash__(self) -> int:
 
+        from jaxsim.utils.wrappers import HashedNumpyArray
+
         return hash(
             (
                 hash(self.name),
                 hash(float(self.mass)),
-                hash(tuple(np.atleast_1d(self.inertia).flatten().tolist())),
-                hash(int(self.index)) if self.index is not None else 0,
-                hash(tuple(np.atleast_1d(self.pose).flatten().tolist())),
+                HashedNumpyArray.hash_of_array(self.inertia),
+                hash(int(self.index)) if self.index is not None else self.index,
```

Comment:
Can we rely on `hash(None)`? This should be reproducible only for Python>=3.12 (see https://github.com/python/cpython/pull/99541)

Reply:
I'm of Python 3.12 and I didn't think about it, if None we can set a 0 hash.